File: digital_insurance_mngmntSystem-main/backend/src/main/java/com/zeta/digital_insurance_management_system/model/Policy.java
package com.zeta.digital_insurance_management_system.model;

import com.zeta.digital_insurance_management_system.enums.Category;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Policy {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String description;
    private BigDecimal premiumAmount;
    private BigDecimal coverageAmount;
    private Integer durationMonths;
    private BigDecimal renewalPremiumRate;
    private LocalDateTime createdAt;

    @Enumerated(EnumType.STRING)
    private Category category;
}

File: digital_insurance_mngmntSystem-main/frontend/src/types/policy.ts
export interface Policy {
    id: number;
    name: string;
    description: string;
    premiumAmount: number;
    coverageAmount: number;
    durationMonths: number;
    renewalPremiumRate: number;
    createdAt?: string | null;
  }
  
  export interface User {
    id: number;
    name: string;
    email: string;
    phone: string;
    address: string;
    role: string;
  }
  
  export interface PurchasedPolicy {
    id: number;
    user: User;
    policy: Policy;
    startDate: string;
    endDate: string;
    status: string;
    premiumPaid: number;
  }